components/people/directory/people_list/people_list.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef, useState} from 'react';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import {FixedSizeGrid, GridChildComponentProps} from 'react-window';
+import InfiniteLoader from 'react-window-infinite-loader';
+
+import {UserProfile} from '@mattermost/types/users';
+
+export interface Props {
+    people: UserProfile[];
+    hasNextPage: boolean;
+    isNextPageLoading: boolean;
+    searchTerms: string;
+    loadMore: () => Promise<void>;
+}
+
+const PeopleList = ({
+    hasNextPage,
+    isNextPageLoading,
+    people,
+    searchTerms,
+    loadMore,
+}: Props) => {
+    const infiniteLoaderRef = useRef<InfiniteLoader | null>(null);
+    const [hasMounted, setHasMounted] = useState(false);
+
+    useEffect(() => {
+        if (hasMounted) {
+            if (infiniteLoaderRef.current) {
+                infiniteLoaderRef.current.resetloadMoreItemsCache();
+            }
+        }
+        setHasMounted(true);
+    }, [searchTerms, people.length, hasMounted]);
+
+    const itemCount = hasNextPage ? people.length + 1 : people.length;
+
+    const loadMoreItems = isNextPageLoading ? () => {} : loadMore;
+
+    const isItemLoaded = (index: number) => {
+        return !hasNextPage || index < people.length;
+    };
+
+    const Item = ({columnIndex, rowIndex, style}: GridChildComponentProps) => {
+        const index = (rowIndex * 4) + columnIndex;
+        if (isItemLoaded(index)) {
+            const user = people[index] as UserProfile;
+            if (!user) {
+                return null;
+            }
+            return (
+                <div
+                    style={style}
+                    key={user.id}
+                >
+                    <>{user.username}</>
+                </div>
+            );
+        }
+
+        return null;
+    };
+
+    if (people.length === 0) {
+        return null;
+    }
+
+    return (
+        <AutoSizer>
+            {({height, width}) => (
+                <InfiniteLoader
+                    ref={infiniteLoaderRef}
+                    isItemLoaded={isItemLoaded}
+                    itemCount={itemCount}
+                    loadMoreItems={loadMoreItems}
+                >
+                    {({onItemsRendered, ref}) => (
+                        <FixedSizeGrid
```

Comment:
Nice!
Just wondering about the column count, will it be somehow dynamically adjusted according to the window width?